src/components/sidenav/sidenav.js

```diff
@@ -257,9 +266,30 @@ function SidenavDirective($mdMedia, $mdUtil, $mdConstant, $mdTheming, $animate,
     scope.$watch('isOpen', updateIsOpen);
 
 
+
     // Publish special accessor for the Controller instance
     sidenavCtrl.$toggleOpen = toggleOpen;
 
+    function revalidateVisibility() {
+      var lastValue = scope.isOpen;
+      if (isHidden(element, true, true) && scope.isOpen == true) scope.isOpen = false;
+      else if (!isHidden(element, true, true) && scope.isOpen == false) scope.isOpen = true;
+
+      if (lastValue != scope.isOpen) {
+        skipSidenav = true;
+        if (!scope.$$phase) scope.$apply();
+        skipSidenav = false;
+      }
+    }
+
+    function isHidden(element, compute, both) {
+      if (!compute || both) {
```

Comment:
In my new version, I renamed to `compute`, it should just allow to check the `offsetParent` and even the `computed style`. I need this, because I check the visibility due the `hide` attributes.